src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,313 @@
+import os
+import logging
+import numpy as np
+from time import sleep
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+try:
+    import clr
+
+    lf_root = os.environ['LIGHTFIELD_ROOT']
+    automation_path = lf_root + '\PrincetonInstruments.LightField.AutomationV4.dll'
+    addin_path = lf_root + '\AddInViews\PrincetonInstruments.LightFieldViewV4.dll'
+    support_path = lf_root + '\PrincetonInstruments.LightFieldAddInSupportServices.dll'
+
+    addin_class = clr.AddReference(addin_path)
+    automation_class = clr.AddReference(automation_path)
+    support_class = clr.AddReference(support_path)
+
+    import PrincetonInstruments.LightField as lf
+
+    clr.AddReference("System.Collections")
+    clr.AddReference("System.IO")
+    from System.Collections.Generic import List
+    from System import String
+    from System.IO import FileAccess
+except Exception as e:
+    logger.error(f"Exception occurred during import: {e}")
+
+
```

Comment:
I agree, we could probably split up the acquire stuff (acquire frame and acquire step and glue) into one class and the settings into another. That is the only split that makes sense to me since the current setup seems to work and is pretty robust but I get if it is more pythonic to change it.

So three total classes? `LightfieldApplicationManager `(which I think we should keep it like it is, but we can place the `acquire `method elsewhere. `SpectrometerSettings` which is self explanatory and `SpectrometerDAQ` which handles all acqusition. That seems like a good idea to me

Reply:
Done